src/main/java/org/truffleruby/core/format/rbsprintf/RBSprintfCompiler.java

```diff
@@ -33,22 +35,30 @@ public RBSprintfCompiler(RubyLanguage language, Node currentNode) {
         this.currentNode = currentNode;
     }
 
-    public RootCallTarget compile(Rope format, Object stringReader) {
-        final RBSprintfSimpleParser parser = new RBSprintfSimpleParser(bytesToChars(format.getBytes()), false);
+    public RootCallTarget compile(Object format, RubyStringLibrary libFormat, TruffleString.GetInternalByteArrayNode byteArrayNode, Object stringReader) {
+        var formatTString = libFormat.getTString(format);
+        var formatEncoding = libFormat.getEncoding(format);
+        var byteArray = byteArrayNode.execute(formatTString, formatEncoding.tencoding);
```

Comment:
Since this is always called behind a TruffleBoundary, there is no need for a GetInternalByteArrayNode, `getInternalByteArrayUncached()` is fine, I'll change to that.